pkg/apis/clusteroperator/v1alpha1/types.go

```diff
@@ -658,3 +660,18 @@ const (
 	// NodeTypeCompute is a node that is a compute node in the cluster
 	NodeTypeCompute NodeType = "Compute"
 )
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CombinedCluster is a cluster object that combines the information from
+// cluster-operator Cluster and cluster-api Cluster.
+type CombinedCluster struct {
```

Comment:
Is this type temporary?

Reply:
Not really. We need some way of keeping the decoded ProviderConfig and ProviderStatus around while we work on the Cluster. So I would expect that we will finalize on something similar to this even after we remove Cluster from clusteroperator.openshift.io.

Reply:
Ok struck me as a little odd given the ClusterAPI* structs already contain the other ClusterOperator* portions, just not decoded.